### PR TITLE
Decentralised Orderbook

### PR DESCRIPTION
**Pull request by rishflab**
The decentralised orderbook component uses libp2p gossipsub to propogate orders to peers. 

Takers can use the take order protocol to take an order published by a maker. A taker sends a take order request to a maker. If the order specified is valid, the maker responds with a confirmation. The maker and taker create a swap and proceed with swap execution. 

Currently only HbitHerc20 swaps are supported where the maker is buying bitcoin and selling erc20 tokens.

### CODE REVIEW

File: api_tests/package-lock.json [hunk 1]
```diff
@@ -0,0 +1,8609 @@
+{
+    "name": "e2e-testing",
```
Comment (thomaseizinger) [inline api_tests/package-lock.json:2]: We are using `yarn` not `npm`, so please delete this file :)

File: cnd/src/trace.rs [hunk 2]
```diff
@@ -10,7 +10,7 @@ pub fn init_tracing(level: log::LevelFilter) -> anyhow::Result<()> {
     }
 
     // We want upstream library log messages, just only at Info level.
-    LogTracer::init_with_filter(LevelFilter::Info)?;
+    LogTracer::init_with_filter(LevelFilter::Debug)?;
```
Comment (thomaseizinger) [inline cnd/src/trace.rs:13]: Change this back to `Info` please.

File: comit/src/network/comit.rs [hunk 3]
```diff
@@ -618,89 +799,93 @@ mod tests {
...
-        // act
-        let (alice_event, bob_event) = future::join(alice_swarm.next(), bob_swarm.next()).await;
-
-        // assert
-        let alice_learned = alice_event.remote_data;
-        let bob_learned = bob_event.remote_data;
-        assert_eq!(bob_learned, want_bob_to_learn_from_alice);
-        assert_eq!(alice_learned, want_alice_to_learn_from_bob);
-    }
+    // todo: fix this test
```
Comment (thomaseizinger) [inline comit/src/network/comit.rs:802]: Please enable this again.

Comment (rishflab) [reply comit/src/network/comit.rs]: fixed

File: cnd/src/http_api/route_factory.rs [hunk 4]
```diff
@@ -136,6 +195,14 @@ pub fn create(facade: Facade, allowed_origins: &AllowedOrigins) -> BoxedFilter<(
...
         .or(hbit_herc20)
         .or(herc20_hbit)
+        .or(take_hbit_herc20_order)
+        .or(get_orders)
+        .or(get_order)
+        .or(make_hbit_herc20_order)
+        .or(post_dial_addr)
+        .or(post_announce_trading_pair)
+        //.or(get_makers)
+        //.or(subscribe)
```
Comment (tcharding) [inline cnd/src/http_api/route_factory.rs:205]: and here

File: cnd/src/main.rs [hunk 5]
```diff
@@ -35,6 +35,7 @@ use rand::rngs::OsRng;
 use std::{process, sync::Arc};
 use structopt::StructOpt;
 use tokio::{net::TcpListener, runtime};
+// use warp::filters::BoxedFilter;
```
Comment (tcharding) [inline cnd/src/main.rs:38]: and here

File: cnd/src/network.rs [hunk 6]
```diff
@@ -129,7 +212,6 @@ fn derive_key_pair(seed: &RootSeed) -> Keypair {
     let key = ed25519::SecretKey::from_bytes(bytes).expect("we always pass 32 bytes");
     Keypair::Ed25519(key.into())
 }
-
```
Comment (tcharding) [inline cnd/src/network.rs:132]: Was this newline removed on purpose?

**Pull request merged**